### DISCUSSION TITLE
can we avoid extra steps needed with automatic testclasspath included

### DISCUSSION THREAD

**Discussion by nagkumar** [qualersoft/jmeter-gradle-plugin#177] [General]
How to make this plugin by default takes all test run time classpath part of the jmeter, without we having to create a fat jar and put in jmeter ext or jmeter junit folder..

**Comment by mathze** [comment]
Not quite sure what you mean. Could you give me more information and maybe a small gh-project showing what you are currently doing? This would help me to better understand the case.

**Comment by nagkumar** [comment]
When we build application all the test and main classes are in build or target directory, does this plugin pickups these classpath directly.. or do we need to package them as fatjar and put it explicitly into 

![image](https://github.com/qualersoft/jmeter-gradle-plugin/assets/332234/1f138a9d-6e27-4eee-8cf4-2d2741cec0a6)


junit folder in lib..

**Reply by mathze** [reply]
No the plugin does not pickup this classpaths. So my first guess would be, that you have to create a fat-jar. But I'm unfamiliar with the `junit` folder and its usage within jmeter, so I can't tell if there would be a better/easier way.

**Comment by nagkumar** [comment]
I see this error

![image](https://github.com/qualersoft/jmeter-gradle-plugin/assets/332234/f750795e-1099-478a-adb7-c194b50c8e5b)

so is there a way to set the default test classpath while running jmeter task using the script
```


    register<JMeterRunTask>("runJMeter") {
        environment(mapOf("JMETER_HOME" to "$buildDir/jmeter"))
        jmxFile.set("TestSuite.jmx")
        generateReport = true
    }
```
